cloud/azure/modules/app/main.tf

```diff
@@ -57,7 +57,7 @@ resource "azurerm_storage_account_network_rules" "files_storage_rules" {
 
 data "azurerm_key_vault" "civiform_key_vault" {
   name                = var.key_vault_name
-  resource_group_name = azurerm_resource_group.rg.name
+  resource_group_name = var.key_vault_resource_group
```

Comment:
why did we add a specific key vault resource group?

Reply:
This is because of a weird circular dependency where we have to create a resource group before we create key vault, but we have to create the key vault before running terraform, which creates a resource group.